### PR TITLE
[EDU-1328] Add Python realtime code samples

### PR DESCRIPTION
**Pull request by m-hulbert**
## Description

This PR adds Python realtime code samples to all relevant locations. It also bumps the Python version to 2.0 and updates the support matrix. All mentions of Python being REST-only have been removed. 

Note: API reference updates will be handled in a separate PR.

## Review

See the review app to view updated content.

### CODE REVIEW

File: content/channels/index.textile [hunk 1]
```diff
@@ -320,6 +336,13 @@ The following is an example of publishing a message to a channel:
   channel.publish 'example', 'message data'
 ```
 
+```[realtime_python]
+  # Python realtime currently utilizes a REST publish
+  realtime = AblyRealtime('{{API_KEY}}')
+  channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}')
+  channel.publish(u'example', u'message data')
```
Comment (owenpearson) [inline content/channels/index.textile:343]:
```suggestion
  await channel.publish(u'example', u'message data')
```

File: content/channels/index.textile [hunk 2]
```diff
@@ -767,6 +798,14 @@ channel.subscribe('myEvent') do |message|
 end
 ```
 
+```[realtime_python]  
+  realtime = AblyRealtime('{{API_KEY}}')
+  channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}')
+  def listener(message):
+    print(f'Message received for {message.name}: ' + message.data)
```
Comment (owenpearson) [inline content/channels/index.textile:805]: string concatenation and interpolation are both fine, but using them both on the same string is a little jarring 😄 
```suggestion
    print(f'Message received for {message.name}: {message.data}')
```

File: content/connect/states.textile [hunk 3]
```diff
@@ -155,6 +161,12 @@ Alternatively a <span lang="default">listener</span><span lang="csharp">handler<
   end
 ```
 
+```[realtime_python]
+  def listener(StateChange):
+      print(StateChange.current)
```
Comment (owenpearson) [inline content/connect/states.textile:166]: pep8 says all variable names should be snake_case
```suggestion
  def listener(state_change):
      print(state_change.current)
```

File: content/getting-started/quickstart.textile [hunk 4]
```diff
@@ -510,12 +526,8 @@ channel.Publish("greeting", "hello!");
...
 
 ```[python] 
-async def main():
-  ably = AblyRest('{{API_KEY}}')
-  channel = ably.channels.get('quickstart')
-  await channel.publish(u'greeting', u'hello!')
-  # other code such as publish, close
-asyncio.run(main())
+channel = ably.channels.get('quickstart')
+await channel.publish(u'greeting', u'hello!')
```
Comment (owenpearson) [inline content/getting-started/quickstart.textile:530]: unicode strings are the default in python 3.0 (ie all currently supported versions of python)
```suggestion
await channel.publish('greeting', 'hello!')
```

**Review by owenpearson — CHANGES_REQUESTED** [review]
Looks good overall! Just a few things that need fixing.

Also, I notice that we're missing docs for listening to channel state changes, I think that ought to be addressed. We're also missing python code examples for authCallback but it seems to be the same story for everything other than ably-js so I guess it's fine to leave that out for now.

File: content/channels/index.textile (hunk 2 above)
Comment (m-hulbert) [reply content/channels/index.textile]: Hah, yeah not sure what I was doing there! Fixed in [6d867e2](https://github.com/ably/docs/pull/1944/commits/6d867e2337199854e6dd60b46e944b748d96f52c)

File: content/getting-started/quickstart.textile (hunk 4 above)
Comment (m-hulbert) [reply content/getting-started/quickstart.textile]: Good to know 👍 Updated in [6d867e2](https://github.com/ably/docs/pull/1944/commits/6d867e2337199854e6dd60b46e944b748d96f52c)

**Review by owenpearson — CHANGES_REQUESTED** [review]
The fixes look good! Still missing channel state change snippets for (1) registering a listener to listen to all channel events, and (2) removing listeners

**Comment by m-hulbert** [comment]
> The fixes look good! Still missing channel state change snippets for (1) registering a listener to listen to all channel events, and (2) removing listeners

You're very right... 🤦 Added in [cfb0737](https://github.com/ably/docs/pull/1944/commits/cfb073754afec1ba74fd8c71ba63f5136e533d02)

**Review by owenpearson — APPROVED** [review]
LGTM 🚀

**Pull request merged**